src/main/webapp/app/lecture/lecture.service.ts

```diff
@@ -114,18 +113,33 @@ export class LectureService {
         return res;
     }
 
-    private checkPermission<ERT extends EntityArrayResponseType>(res: ERT): ERT {
+    private setAccessRightsLectureEntityArrayResponseType<ERT extends EntityArrayResponseType>(res: ERT): ERT {
         if (res.body) {
             res.body.forEach((lecture: Lecture) => {
-                if (lecture.course) {
-                    lecture.isAtLeastInstructor = this.accountService.isAtLeastInstructorInCourse(lecture.course);
-                    lecture.isAtLeastEditor = this.accountService.isAtLeastEditorInCourse(lecture.course);
-                }
+                this.setAccessRightsLecture(lecture);
             });
         }
         return res;
     }
 
+    /**
+     * Besides the within the lecture included variables for access rights the access rights of the
+     * respective course are set aswell.
+     *
+     * @param lecture for which the access rights shall be set
+     * @private
+     */
+    private setAccessRightsLecture(lecture: Lecture | null) {
```

Comment:
[nit] you forgot the `@return` tag. But frankly, do we even need to return the `Lecture` in this method?
Can we make this method, be a method of class `Lecture`? As in:
```
class Lecture {
  ...
  setAccessRights() {
    ...
  }
}
```

Reply:
Thanks for the review, the nitpick about the comment is fixed in https://github.com/ls1intum/Artemis/pull/4201/commits/5a069000921ad665e0170c134a70481c1d31fb53
Within this service the return value is not needed yet, it is done like that within other services so I decided to keep it consistent. 
I would not put this method into the class itself, as we are currently using those classes rather as models as far as I know. I think that the `AccountService` would be a place better place for such a method.

Reply:
When refactoring this, the other services should be touched aswell _(services that have been touched in https://github.com/ls1intum/Artemis/pull/3742)_